web-client/src/ustc-ui/DateInput/DateSelector.tsx

```diff
@@ -40,19 +43,28 @@ export const DateSelector = ({
        * we can get the actual visible value of the input.
        */
       datePicker.on(formGroupInputRef.current);
-      const myDatePicker = formGroupInputRef.current.querySelector(
-        `#${id}-picker.usa-date-picker__external-input`,
-      );
+      const myDatePicker =
+        formGroupInputRef.current.querySelector(datePickerId);
+
       if (!myDatePicker) throw new Error('could not find expected date picker');
-      myDatePicker.addEventListener('change', onChange);
-      myDatePicker.addEventListener('input', onChange);
+
+      (myDatePicker as HTMLInputElement).addEventListener('change', onChange);
+      (myDatePicker as HTMLInputElement).addEventListener('input', onChange);
     }
   }, [formGroupInputRef]);
 
   useEffect(() => {
     if (formGroupInputRef.current) {
       const input = formGroupInputRef.current.querySelector('.usa-date-picker');
       if (disabled) {
+        const myDatePicker =
+          formGroupInputRef.current.querySelector(datePickerId);
+
+        if (!myDatePicker)
+          throw new Error('could not find expected date picker');
+
+        (myDatePicker as HTMLInputElement).value = '';
```

Comment:
Out of curiosity only, is there a reason we have to do this instead of giving the datepicker a default value of ''?

Reply:
When you apply a stamp to an order, there is an option to 'Clear Optional Fields'. One of the optional fields happens to have a date selector inside of it. This was the cleanest way I could figure out to reset the date selector value because of how USWDS exposes/handles the date picker value.

Reply:
Makes sense, thank you!